enhancements/storage/storage-hypershift.md

```diff
@@ -0,0 +1,1050 @@
+      * Add `token-minter` sidecar, which will create STS secrets to the driver.
+      * Set `priorityClassName: hypershift-control-plane`.
+    * Phase 2: add metrics services + TLS keys.
+
+* **AWS EBS CSI driver**:
+  * Phase 1: no code changes.
+    * Controller parts will run in the management cluster, but they get kubeconfig to the guest cluster.
+    * Node DaemonSet will get the default projected kubeconfig that can access objects only in the
+    openshift-cluster-csi-drivers namespace.
+  * Phase 2 (or even later): reduce memory requirements of AWS EBS CSI driver control-plane Deployment.
+    * Review memory requirements of each sidecar.
+    * Union kube-rbac-proxies, if possible.
+    * Union all sidecars into a single container, so they share informers? This will need serious investment upstream.
+  * Phase 2 (or even later): move leader-election of CSI sidecars to the management cluster.
+    * I.e. the sidecars get a special kubeconfig just for the leader election.
+    * The sidecars already have a parameter with namespace where the Lease objects are created, CSI driver operators
+      will need to set correctly in HyperShift, so CSI sidecars of different guest clusters don't collide.
+
+#### Notes about other CSI drivers
+
+* Azure Disk CSI driver
+  * It needs some cloud credentials available on nodes. We need to check what permissions the CSI driver pods that run
+    on worker nodes actually need and perhaps create two roles in Azure - one for the control-plane of the CSI driver
+    (create / delete / attach / detach volumes) and second for the node (read volume details?). 
+    
+* vSphere CSI driver
+  * Controller parts of the CSI driver assume that they run in vSphere. Significant driver changes may be needed
+    to be able to run the controller parts on any cloud or on bare metal.
+  * vSphere is often behind a firewall and is often not accessible from outside of vSphere itself. I.e. if the
+    management cluster is outside of vSphere, it may be necessary to open new firewall ports.  
+* On vSphere, cluster-storage-operator installs vsphere-problem-detector. It's not clear if it should run in the
+  management cluster or in the guest cluster or in both. Because both cluster will need to talk to vSphere API
+  and the problem detector is there to check for common problems in the connection.
+
+### Risks and Mitigations
+
+#### Storage CRs are in the guest cluster
+
+* Since CRs for cluster-storage-operator, aws-ebs-csi-driver-operator and cluster-csi-snapshot-controller-operator are
+  in the guest cluster, a Cluster Instance Admin can DOS the whole CSI driver / CSI snapshots by editing these CRs in
+  an endless loop. The operators will redeploy the operands with each CR change and the new operands will need to wait
+  for leader election to expire, while the old operands are already terminated.
+  * We mitigate this issue a bit by overwriting user changes in the operator CRs by Hosted Cluster Config Operator that
```

Comment:
We could use a validating webhook to prevent this. But in general, there are lots of ways for a cluster admin to break their cluster, if they explicitly attempt to.